dotCMS/src/main/java/com/dotcms/util/RelationshipUtil.java

```diff
@@ -82,21 +82,28 @@ public static List<Contentlet> filterContentlet(final long language, final Strin
         //LinkedHashMap to preserve the order of the contentlets
         final Map<String, Contentlet> relatedContentlets = new LinkedHashMap<>();
 
+        boolean isLuceneQuery = UtilMethods.isLuceneQuery(filter);
+
         //Filter can be an identifier or a lucene query (comma separated)
         for (final String elem : filter.split(StringPool.COMMA)) {
-            if (UUIDUtil.isUUID(elem.trim()) && !relatedContentlets.containsKey(elem.trim())) {
-                final Identifier identifier = identifierAPI.find(elem.trim());
-                final Contentlet relatedContentlet = contentletAPI
-                        .findContentletForLanguage(language, identifier);
-                relatedContentlets.put(relatedContentlet.getIdentifier(), isCheckout ? contentletAPI
-                        .checkout(relatedContentlet.getInode(), user, respectFrontendRoles)
-                        : relatedContentlet);
-            } else {
-                relatedContentlets
-                        .putAll((isCheckout ? contentletAPI.checkoutWithQuery(elem, user, false)
-                                : contentletAPI.search(elem, 0, 0, sortBy, user, false)).stream()
-                                .collect(Collectors
-                                        .toMap(Contentlet::getIdentifier, Function.identity(),(oldValue, newValue) -> oldValue)));
+            if (!filter.isEmpty()) {
+                if (!UUIDUtil.isUUID(elem.trim()) && !isLuceneQuery) {
```

Comment:
We are already checking if `elem.trim()` is a UUID in two places.
Maybe you don't need to check it in line 90.
But if you must, then store the result of  `UUIDUtil.isUUID(elem.trim())`  in a boolean variable.

Reply:
agree in the worst case the method will be called twice

final boolean isUUID = UUIDUtil.isUUID(elem.trim());